internal/controller/pkg/revision/reconciler.go

```diff
@@ -226,13 +229,29 @@ func SetupProviderRevision(mgr ctrl.Manager, o controller.Options) error {
 		return errors.Wrap(err, "cannot build fetcher for package parser")
 	}
 
+	var hookOpts []ProviderHookOption
+	if o.Features.Enabled(features.EnableAlphaExternalSecretStores) {
+		hookOpts = []ProviderHookOption{
+			WithControllerEnvironments([]corev1.EnvVar{
+				{
+					Name:  "POD_NAMESPACE",
+					Value: o.Namespace,
+				},
+				{
+					Name:  "ENABLE_EXTERNAL_SECRET_STORES",
```

Comment:
There are a couple of options to enable an alpha feature for providers, external secret store in this case:

a. Do nothing in Crossplane core, expect users to create a ControllerConfig for each provider with `--enable-external-secret-stores` argument.
b. If the feature is enabled for core Crossplane, pass it as an argument to any provider installed.
c. If the feature is enabled for core Crossplane, pass it as an environment variable to any provider installed.

Option a would require documentation and additional work which could be automated.
Option b would cause a provider controller to fail with an unknown argument error if the feature is not supported (yet).
Option c seems like the best option UX wise that I could think of. The only caveat seems to be not being explicit in terms of whether the feature is not enabled or not working properly. I am planning to implement a `disabled` connection details manager included in the connection publishers if the feature is not enabled, which will simply return a proper error whenever the feature is not enabled but tried to be used by configuring `publishConnectionDetailsTo` API.

Reply:
Hrm, for an alpha feature I personally prefer option A. I imagine as soon as this feature graduates from alpha this we'll default `--enable-external-secret-stores=true` and this won't be a problem anymore.

Reply:
I am happy to go in both directions. Given this is an alpha feature, a bit of additional work wouldn't hurt as long as we have documented.

Reply:
Updated according to option a.